src/main/java/org/folio/processing/mapping/mapper/writer/marc/MarcBibRecordModifier.java

```diff
@@ -27,30 +31,32 @@ public class MarcBibRecordModifier extends MarcRecordModifier {
     "611", "630", "700", "710", "711", "730", "800", "810", "811", "830");
 
   private List<Link> bibAuthorityLinks = emptyList();
+  private List<LinkingRuleDto> linkingRules = emptyList();
   private final Set<Link> bibAuthorityLinksKept = new HashSet<>();
 
   public List<Link> getBibAuthorityLinksKept() {
     return new LinkedList<>(bibAuthorityLinksKept);
   }
 
   public void initialize(DataImportEventPayload eventPayload, MappingParameters mappingParameters,
-                         MappingProfile mappingProfile, EntityType marcType,
-                         InstanceLinkDtoCollection links) throws IOException {
+    MappingProfile mappingProfile, EntityType marcType,
+    InstanceLinkDtoCollection links, List<LinkingRuleDto> linkingRules) throws IOException {
     validateEntityType(marcType);
     initialize(eventPayload, mappingParameters, mappingProfile, marcType);
     if (links == null || links.getLinks() == null) {
       return;
     }
     bibAuthorityLinks = links.getLinks();
+    this.linkingRules = linkingRules;
   }
 
   /**
    * Should call regular update subfield flow only for uncontrolled subfields
-   * */
+   */
   @Override
   protected boolean updateSubfields(String subfieldCode, List<DataField> tmpFields, DataField fieldToUpdate,
-                                    DataField fieldReplacement, boolean ifNewDataShouldBeAdded) {
-    var linkOptional = getLink(fieldToUpdate);
+    DataField fieldReplacement, boolean ifNewDataShouldBeAdded) {
+    var linkOptional = getLink(fieldReplacement);
```

Comment:
You need to use 'fieldToUpdate' here, because incoming could be modified and no longer linked